src/query/procedure/mg_procedure_impl.hpp

```diff
@@ -675,8 +675,27 @@ struct mgp_vertices_iterator {
   /// @throw anything VerticesIterable may throw
   mgp_vertices_iterator(mgp_graph *graph, memgraph::utils::MemoryResource *memory)
       : memory(memory), graph(graph), vertices(graph->impl->Vertices(graph->view)), current_it(vertices.begin()) {
-    if (current_it != vertices.end()) {
+    memgraph::utils::OnScopeExit clean_up([this] { this->current_v = std::nullopt; });
```

Comment:
What about `mgp_edge_get_from` and `mgp_edge_get_to`?

Reply:
This is something that we are addressing in one of the next PRs